backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/ProposalUnsafe.kt

```diff
@@ -0,0 +1,29 @@
+package cash.z.ecc.android.sdk.internal.model
+
+import cash.z.wallet.sdk.ffi.ProposalOuterClass.FeeRule
+import cash.z.wallet.sdk.ffi.ProposalOuterClass.Proposal
+
+class ProposalUnsafe(
+    private val inner: Proposal
+) {
+    init {
+        require(inner.feeRule != FeeRule.FeeRuleNotSpecified) {
+            "Fee rule must be specified"
+        }
+    }
+
+    companion object {
+        fun parse(encoded: ByteArray): ProposalUnsafe {
+            val inner = Proposal.parseFrom(encoded)
```

Comment:
Isn't protobuf parsing fallible? How is such an error represented here, is it thrown? Is there no way of indicating that in the type signature?